apps/task/models.py

```diff
@@ -45,9 +45,22 @@ def __str__(self):
     @staticmethod
     def get_for(user):
         return TaskGroup.objects.filter(
-            user_group__members=user
+            users__in=[user]
         ).distinct()
 
+    def duration_taskgroup(self, user):
+        task_group = TaskGroup.objects.filter(id=self.id).annotate(
+            duration=models.Subquery(
+                Task.objects.order_by().filter(
+                    task_group=models.OuterRef('pk'),
+                    user=user
+                ).values('task_group').annotate(
+                    task_duration=models.Sum(F('timeentry__end_time') - F('timeentry__start_time'))
+                ).values('task_duration')
+            )
+        ).aggregate(sum=models.Sum('duration'))['sum']
+        return task_group
+
```

Comment:
```suggestion
    def duration_taskgroup(self, user):
                total_duration = Task.objects.order_by().filter(
                    task_group=self.id,
                    user=user
                ).values('task_group').annotate(
                    task_duration=models.Sum(F('timeentry__end_time') - F('timeentry__start_time'))
                ).values('task_duration')
            )
        return total_duration

```

Reply:
Maybe something like this?

Reply:
Also, not sure how this is used but try to avoid the n+1 problem.